lab7/app/src/main/java/com/example/project2/MainActivity.java

```diff
@@ -0,0 +1,146 @@
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import java.io.IOException;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.KeyStore;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
+import java.security.NoSuchProviderException;
+import java.security.PrivateKey;
+import java.security.PublicKey;
+import java.security.SecureRandom;
+import java.security.cert.CertificateException;
+import java.util.Base64;
+import java.util.Enumeration;
+
+import javax.crypto.Cipher;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+public class MainActivity extends AppCompatActivity implements View.OnClickListener {
+    public static final String SHARED_PREFS = "sharedPrefs";
+    public Button savebtn,showpinbtn;
+    public KeyPair pair;
+    public EditText textbox;
+    public TextView status;
+    public  KeyStore ks;
+    @Override
+    protected void onCreate(Bundle savedInstanceState)  {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        KeyguardManager km = (KeyguardManager)getApplicationContext().getSystemService(Context.KEYGUARD_SERVICE);
+        assert km != null;
+        if(km.isKeyguardSecure())
+            Toast.makeText(getApplicationContext(), "Phone is Secure!", Toast.LENGTH_LONG).show();
+        else {
+            Toast.makeText(getApplicationContext(), "Phone is Unlocked!", Toast.LENGTH_LONG).show();
+            finish();
+        }
+        try {
+            ks = KeyStore.getInstance("AndroidKeyStore");
+            ks.load(null);
+            Enumeration<String> aliases = ks.aliases();
+            pair = generateKeyPair();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+        savebtn=findViewById(R.id.save);
+        showpinbtn=findViewById(R.id.show_pin);
+        showpinbtn.setOnClickListener(this);
+        textbox=findViewById(R.id.textbox);
+        savebtn.setOnClickListener(this);
+        status=findViewById(R.id.status);
+    }
+
+
+    public static KeyPair generateKeyPair() throws Exception{
+        String alias="rsakey1";
+        KeyPairGenerator generator = KeyPairGenerator.getInstance("RSA", "AndroidKeyStore");
+        KeyGenParameterSpec keyGenParameterSpec = new KeyGenParameterSpec.Builder(alias, KeyProperties.PURPOSE_ENCRYPT | KeyProperties.PURPOSE_DECRYPT).setKeySize(2048).setEncryptionPaddings(KeyProperties.ENCRYPTION_PADDING_RSA_PKCS1).build();
+        generator.initialize(keyGenParameterSpec);
+        return generator.generateKeyPair();
+    }
+
+    public void saveData(String text){
+        SharedPreferences sharedPreferences =getSharedPreferences(SHARED_PREFS,MODE_PRIVATE);
+        SharedPreferences.Editor editor = sharedPreferences.edit();
+        editor.putString("encryptedtext",text);
```

Comment:
You should keep such strings as constants instead of hardcoding them. Hardcoding values is also error (typo) prone.